rust/ql/integration-tests/macro-expansion/test.expected

```diff
@@ -0,0 +1,11 @@
+| src/lib.rs:3:1:4:11 | fn foo | 0 | src/lib.rs:4:1:4:10 | fn foo_0 |
+| src/lib.rs:3:1:4:11 | fn foo | 1 | src/lib.rs:4:1:4:10 | fn foo_1 |
+| src/lib.rs:3:1:4:11 | fn foo | 2 | src/lib.rs:4:1:4:10 | fn foo_2 |
+| src/lib.rs:6:1:8:11 | fn bar | 0 | src/lib.rs:7:1:8:10 | fn bar_0 |
+| src/lib.rs:6:1:8:11 | fn bar | 1 | src/lib.rs:7:1:8:10 | fn bar_1 |
```

Comment:
I had not expected `fn bar_0` and `fn bar_1` here. Shouldn't the expansion be as follows? 

```
#[repeat(2)]
#[repeat(3)]
fn bar() {}
```

-->

```
#[repeat(3)]
fn bar_0() {}
#[repeat(3)]
fn bar_1() {}
```

-->

```
fn bar_0_1() {}
fn bar_0_2() {}
fn bar_0_3() {}
fn bar_1_1() {}
fn bar_1_2() {}
fn bar_1_3() {}
```

Reply:
indeed, that is the expansion, but all steps of the expansion are in the database. So you get the immediate expansion `bar_0` and `bar_1` with their `repeat` attribute attached to them, and then you get `bar_i_j` as `getAttributeMacroExpansion()` under each `bar_i`

Reply:
this is similar to what we already do with macro call expansions: we don't get the full expansion right away, and each macro call gets in turn expanded within the expansion of the top-level macro call

Reply:
Perfect , that makes sense. I was just wondering.